tensorboard/plugins/profile/pod_viewer/stack_bar_chart/stack-bar-chart.ts

```diff
@@ -0,0 +1,203 @@
+/* Copyright 2015 The TensorFlow Authors. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the 'License');
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+namespace pod_viewer_stack_bar_chart {
+
+Polymer({
+  is: 'stack-bar-chart',
+  properties: {
+    id: {
+      type: String,
+    },
+    data: {
+      type: Object,
+      notify: true,
+      observer: 'dataChanged_',
+    },
+    ready_: {
+      type: Boolean,
+      value: false,
+    },
+    stackEle: {
+      type: Array,
+      notify: true,
+      observer: 'onStackEleChanged_',
+    },
+    xDomainFunc: {
+      type: Object,
+      notify: true,
+    },
+    active: {
+      type: Object,
+      notify: true,
+    },
+  },
+  /**
+   * Main function to draw a stacked bar chart.
+   */
+  stackBarChart: function(data) {
+    if (!data || !this.ready_ || this.stackEle.length == 0) {
+      return;
+    }
+    d3.selectAll('#' + this.id + ' g > *').remove();
```

Comment:
This is a global querySelector. `d3.select(node)` is allowed but `d3.select('.querySelector')` breaks badly on real WebComponent. Please use ```d3.select(this).select(`#${this.id} g > *`)```. Here and below.

Reply:
Done.